skafka/src/main/scala/com/evolutiongaming/skafka/consumer/ConsumerConverters.scala

```diff
@@ -89,6 +90,52 @@ object ConsumerConverters {
     }
   }
 
+  implicit class RebalanceListener1Ops[F[_]](val self: RebalanceListener1[F]) extends AnyVal {
+
+    def asJava(consumer: ConsumerJ[_, _])(implicit
+      F: Concurrent[F],
+      toTry: ToTry[F],
+    ): RebalanceListenerJ = {
+
+      def onPartitions(
+        partitions: CollectionJ[TopicPartitionJ],
+        call: Nes[TopicPartition] => RebalanceCallback[F, Unit]
+      ) = {
+        val result = partitions
+          .asScala
+          .toList
+          .traverse {_.asScala[F]}
+          .map { partitions =>
+            partitions
+              .toSortedSet
+              .toNes
+          }
+          .toTry
+          .flatMap { nesOpt =>
+            nesOpt.map {
+              nes => RebalanceCallback.run(call(nes), consumer)
+            }.getOrElse(Try(()))
```

Comment:
().pure[Try] :)

Reply:
good idea, thanks! fixed in bbd8c114c0e5a9efeb82e1f338b7055c26eb57b7